Generator/AdapterGenerator.cs

```diff
@@ -0,0 +1,298 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace NodeApi.Generator;
+
+/// <summary>
+/// Generates C# <-> JavaScript adapter code for C# members exported to JavaScript.
+/// </summary>
+internal class AdapterGenerator : SourceGenerator
+{
+    private const string AdapterPrefix = "__";
```

Comment:
>private const string AdapterPrefix = "__"; [](http://example.com/codeflow?start=4&length=42)

I am just curious why to prefix all types and variables with the double underscore? Can we just put them all in a special namespace?

Reply:
Types are not prefixed, only generated method names and variables in those methods.

Adapter method names include the full type name of the adapted method to avoid conflicts when multiple classes have the same method name. But they don't really need the extra `__` prefix since they are in a separately generated class.

I had considered generating the adapter methods in the same class as the target methods. That would require the class to be marked `partial`, similar to how `[LibraryImport]` generated methods work. Then the `__` or similar prefix or suffix would be needed. I didn't choose that approach because I didn't want to require modifying the original code.

The variables in generated code are prefixed with `__` to avoid any conflicts with the original method parameter names. in case the user's method has parameters named `args` or `result`. That prefix is a convention for generated code, and `[LibraryImport]` generated code does the same thing.

Reply:
Updated.